src/server/saved_objects/service/lib/version.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { createInvalidVersionError } from './errors';
+
+const decodeBase64 = (base64: string) => Buffer.from(base64, 'base64').toString('utf8');
+const encodeBase64 = (utf8: string) => Buffer.from(utf8, 'utf8').toString('base64');
+
+/**
+ * Encode the sequence params into an "opaque" version string
+ * that can be used in the saved object API in place of numeric
+ * version numbers
+ */
+export function encodeVersion(seqNo: number, primaryTerm: number) {
+  if (!Number.isInteger(primaryTerm)) {
+    throw new TypeError('_primary_term from elasticsearch must be an integer');
+  }
+
+  if (!Number.isInteger(seqNo)) {
+    throw new TypeError('_seq_no from elasticsearch must be an integer');
+  }
+
+  return encodeBase64(JSON.stringify([seqNo, primaryTerm]));
```

Comment:
Why JSON.stringify over simple concatenation?

Reply:
So it's easy to decode without having to write a new encoding format

Reply:
Looks like using JSON is 66% slower to encode, but 5% faster to decode. Will leave it up to you to determine if the tradeoff is worth it.

Reply:
Idk, I did some testing locally and I'm getting inconsistent results but I'm kinda religious about not inventing new formats and the times I'm measuring are very tiny so I'm not convinced it's worth it to drill super deep and optimize here.